src/main/webapp/app/course/dashboards/assessment-dashboard/exam-assessment-buttons/exam-assessment-buttons.component.ts

```diff
@@ -64,18 +64,24 @@ export class ExamAssessmentButtonsComponent implements OnInit {
             this.courseService.find(this.courseId).subscribe((courseResponse) => {
                 this.course = courseResponse.body!;
             });
-            const studentExamObservable = this.studentExamService.findAllForExam(this.courseId, this.examId).pipe(
-                tap((res) => {
-                    // TODO: directly load the longest working time from the server and avoid loading all student exams
-                    this.setStudentExams(res.body);
-                    this.longestWorkingTime = Math.max.apply(
-                        null,
-                        this.studentExams.map((studentExam) => studentExam.workingTime),
-                    );
+
+            /*
+            Prepare workingTimeObservable to perform the following on subscribe:
+            - set the longestWorkingTime
+            - trigger (re)calculation of whether the exam is over
+             */
```

Comment:
This is a personal opinion, but I do think that the comments are not needed - for me it seems like a reformulation of the code that follows, which to me, seems to be easy to understand and does not need further explanation.

For me, the comment doesn't add information 😅

Reply:
You got a fair point with the last two lines of comment, but personally I think that the first line provides value. It describes the fact that the following code does not actually retrieve any code but rather only modifies the "query" for later evaluation. Taking that into consideration, I´d argue that the last two lines are required to distinguish the two query-preparation statements. Dont you think so?